### DISCUSSION TITLE
GET_X_LPARAM

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [microsoft/CsWin32#744] [Q&A] [answered]
Where are these function for accessing mouse coordinates?

https://learn.microsoft.com/en-us/windows/win32/api/windowsx/nf-windowsx-get_x_lparam

Would this function posted by Jimi on StackOverflow do the job?
```cs
    private static (int x, int y) PointFromLParam(IntPtr lParam)
    {
        return ((int)(lParam) & 0xFFFF, ((int)(lParam) >> 16) & 0xFFFF);
    }
```

**Comment by AArnott** [comment, marked as answer]
It looks like `GET_X_LPARAM` is a macro. Macros are a feature of the Windows SDK unique to the C/C++ consumers and are not available in the metadata, and thus won't generally be emitted by CsWin32.
In such cases, manual translation from the C++ definition to C# as you suggest is the way to go.
I am not familiar with `GET_X_LPARAM` in particular, so I can't comment on the efficacy of your `PointFromLParam` method.

**Reply by Gavin-Williams** [reply]
Thanks for explaining that.  And the given function seems to work so far.
